### ISSUE TITLE
question about training stft

### ISSUE THREAD

**Issue by starLitEE** [Rikorose/DeepFilterNet#222]
Hi. Thank you for your code.
Now I’m on the process of converting your rust stft code to python.
I have several questions about your data loading and stft process.
1.	On training process, the datasets are loaded with torch.audio without normalization option. As I checked, after some process of stft (through rust code), the output value was normalized. Does is processed by multiplying window norm factor?
2.	On enhancement process, the audio is loaded with librosa with normalizing option. The loaded audio is transformed to stft with rust code. This stft transformation process is exactly the same as the stft process on training process?  Can you explain about difference between analysis function and stft function in your rust code?
3.   Additionaly, about zero padding for time domain signal before applying window function. Torch audio apply reflection pading to left and right side of signal. How about your code? Adding zero only for the first part of the signal is right? 

Thankyou for your answering!

**Comment by Rikorose** [comment]
1. Normalization in torchaudio.load means int16 to f32 conversion and is a different thing than the moving average normailzation of the input features
2. Training and inference stft are the same. During training the stft and input features are computed within the dataloader framework. Which functions do you mean exactly?
3. Reflection center padding is not real time compliant, zero padding first part is right.

**Comment by starLitEE** [comment]
Thank you for your answer.

For 2, I was meaning fn stft and fn analysis in rust code. Are they works equal?

**Comment by Rikorose** [comment]
> For 2, I was meaning fn stft and fn analysis in rust code. Are they works equal?

Yes

We mix one speech signal with multiple superimposed noise signals resulting in one noisy mixture.

**Comment by starLitEE** [comment]
Really helpful. Thank you very much.
